File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include <set>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    set<int> s;
    
    for(int i = 0; i < numbers.size(); i++){
        
        int a = numbers[i];
        for(int j = i + 1; j < numbers.size(); j++){
            int b = numbers[j];
            s.insert(a+b);
        }
    }
    
    for(auto a : s){
        answer.push_back(a);
    }
    
    //sort(answer.begin(), answer.end());
    return answer;
}

File: Java/프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = {};
        List<Integer> arr = new ArrayList<>();
    
        for(int i=0; i<numbers.length; i++){
            for(int j=i+1; j<numbers.length; j++){
                int sum = numbers[i] + numbers[j];
                if(!arr.contains(sum)){
                    arr.add(sum);
                }
            }
        }
        
        Collections.sort(arr);
        
        answer = new int[arr.size()];
        for(int i=0; i<arr.size(); i++){
            answer[i] = arr.get(i);
        }
        
        return answer;
    }
}